src/main/java/org/junit/Assert.java

```diff
@@ -955,4 +955,48 @@ public static void assertEquals(Object[] expecteds, Object[] actuals) {
             Matcher<? super T> matcher) {
         MatcherAssert.assertThat(reason, actual, matcher);
     }
+
+    /**
+     * This interface facilitates the use of expectThrows from Java 8. It allows method references
+     * to void methods (that declare checked exceptions) to be passed directly into expectThrows
+     * without wrapping. It is not meant to be implemented directly.
+     *
+     * @since 4.13
+     */
+    public interface ThrowingRunnable {
+        void run() throws Throwable;
+    }
+
+    /**
+     * Asserts that {@code runnable} throws an exception of type {@code throwableClass} when
+     * executed. If it does, the exception object is returned. If it does not throw an exception, an
+     * {@link AssertionError} is thrown. If it throws the wrong type of exception, an {@code
+     * AssertionError} is thrown describing the mismatch; the exception that was actually thrown can
+     * be obtained by calling {@link AssertionError#getCause}.
+     *
+     * @param throwableClass the expected type of the exception
+     * @param runnable       A function that is expected to throw an exception when executed
+     * @return The exception thrown by {@code runnable}
+     * @since 4.13
+     */
+    public static <T extends Throwable> T expectThrows(Class<T> throwableClass, ThrowingRunnable runnable) {
+        try {
+            runnable.run();
+        } catch (Throwable t) {
+            if (throwableClass.isInstance(t)) {
```

Comment:
> Your original pull request was opened only two days ago, and we responded on the same day, so I think it may be a bit of a stretch to say we are dragging our feet :-)

I don't think it is. There was [a nearly identical PR](https://github.com/junit-team/junit/pull/1055) _half a year ago_, and it went nowhere. And there was a [detailed proposal](https://github.com/junit-team/junit/issues/706) six months before _that_. The scary thing about #1055 is that it was closed because a strange consensus was reached that the functionality belonged in junit-lambda, at least initially. Six months later, junit-lambda has gone absolutely nowhere. But my point is that the maintainers have had over a year to meditate on this design. It would not be an act of reckless hastiness to merge this PR as-is.

> Secondly, I ask that you please have patience in our process.

That would be easier if there were a clear process. That's what I was trying to tell you when I responded to your remark that "[o]nce we add an API to JUnit, it's hard to remove it, but it's possible we could have some Beta APIs JUnit-Lambda (possibly using an annotation like Guava's @Beta annotation)." You're basically telling me that an open source project that has been around since the Clinton administration isn't sure how to _add a simple feature_. Why is this not a solved problem? Can you see how your remark can make the JUnit team look rudderless and stagnant?

Keep in mind that I've been very thorough in addressing feedback, either incorporating it or responding with (very solid, I think) counterarguments. Since I haven't heard any new feedback in a while, I'm left sitting here wondering what the problem is. What is everybody waiting for?

> Designing an API takes time. The size of the API is often unrelated to the amount of time it might take to ensure it's what we want to build, and make sure it's polished and well documented.

There's a difference between design and analysis paralysis. Seeing all of the old discarded PRs for this feature (which I only discovered recently) makes me worry that this discussion is tending towards the latter.